utilities/load-test/loadtest/loadtest_test.go

```diff
@@ -2,38 +2,355 @@ package loadtest
+	"reflect"
 	"testing"
+	"time"
 
-	// appv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	appv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func TestAThing(t *testing.T) {
+// type appResource struct{
+// 	AppInfo *appv1.Application
+// 	Count int
+// }
 
-	// GetE2EEFixtureK8sClient is from the ApplicationSet repo.
-	// You can see more examples of how to use it, here: https://github.com/argoproj-labs/applicationset/search?q=GetE2EFixtureK8sClient
+var (
+	// Check for an API (For now it is asume that the kubeconfig og each user lies in ~/.kube/config)
+	userLocal, _ = user.Current()
+	kubeconfig   = userLocal.HomeDir + "/.kube/config"
+	master       = ""
+	namespace    = "argocd"
+	podName      = ""
+)
+
+func TestNGuestbook(t *testing.T) {
+	// TEST 1: Deply 100 guestbook in different namespace
+	fmt.Print("Testing for 100 Guestbook applications\n")
 
-	// In this example, I am retrieving all Application CRs within the 'argocd' namespace
+	fmt.Print("\n\nPods Memory Resource before creation:\n\n")
+	memoryInit := Get_pod_info(kubeconfig, master, namespace, podName)
+	PodInfoParse(memoryInit)
 
-	appList, err := GetE2EFixtureK8sClient().AppClientset.ArgoprojV1alpha1().Applications("argocd").List(context.TODO(), v1.ListOptions{})
+	// Get Pod Restart Count
+	PodC := PodRestartcount(namespace)
+
+	for i := 1; i < 101; i++ {
+		guestbookApp := &appv1.Application{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "Application",
+				APIVersion: "argoproj.io/v1alpha1",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "guestbook-" + fmt.Sprintf("%d", i),
+				Namespace: "argocd",
+			},
+			Spec: appv1.ApplicationSpec{
+				Source: appv1.ApplicationSource{
+					Path:           "guestbook",
+					TargetRevision: "HEAD",
+					RepoURL:        "https://github.com/argoproj/argocd-example-apps.git",
+				},
+				Destination: appv1.ApplicationDestination{
+					Namespace: "guestbook-" + fmt.Sprintf("%d", i),
+					Server:    "https://kubernetes.default.svc",
+				},
+				Project: "default",
+			},
+		}
+
+		_, err_create := GetE2EFixtureK8sClient().AppClientset.ArgoprojV1alpha1().Applications(namespace).Create(context.TODO(), guestbookApp, v1.CreateOptions{})
+		if err_create != nil {
+			t.Errorf("error, %v", err_create)
+			return
+		}
+	}
 
-	if err != nil {
-		t.Errorf("error, %v", err)
+	// Check if the Pod is getting refreshed or not
+	if reflect.DeepEqual(PodC, PodRestartcount(namespace)) != true {
+		t.Errorf("error, %v", "The Pods are getting refreshed!")
 		return
 	}
 
-	for _, app := range appList.Items {
+	fmt.Print("\n\nPods Memory Resource after Creation:\n\n")
+	memoryPost := Get_pod_info(kubeconfig, master, namespace, podName)
+	PodInfoParse(memoryPost)
+
+	fmt.Print("\n\nDifference in the Pod Memory Usage (in Ki)\n\n")
+	delta := PodMemoryDiff(memoryInit, memoryPost)
+	PodInfoParse(delta)
+
+	// less use of generic sleep statements in automated tests
```

Comment:
I've found the example I wanted to give you, which is a way to avoid using sleep statements in tests:
https://github.com/redhat-developer/gitops-operator/blob/d9b4c4844b39f78683cd30525993e9af19929069/test/helper/application_status.go#L164

(this is a pattern I recommend using for all test code, as an alternative to use sleep statements)

Ultimately it looks like this:
```go
err := wait.Poll( /* time to wait between retries*', /* max time to wait */, func() (bool, err) 

    // return true if the thing we were waiting for has happened

    // if an error occurs, 'return false, error', which will end the loop.

    return false // this will keep the loop going, until true is returned

}) 
if err != nil {
   // fail the test
}
```

So `wait.Poll` will keep checking whether a particular condition is met, and will timeout with an error it is not. 

In this case, we probably want to wait for all the Argo CD applications to be processed by Argo CD, before check the pod memory usage.

In order to tell if Argo CD has finished processing the Application, we can look in the `(app variable).Status.Health.Status` field, which will be empty (`""`) if the Argo CD hasn't yet processed the Application resource.

So that might look like this:
```go
err := wait.Poll( /* time to wait between retries*', /* max time to wait */, func() (bool, err) {

// list all Argo CD applications in the namespace
    // for each application, check if (app variable).Status.Health.Status
        // if status is empty, return false

return true
})
```
So this will keep waiting until Argo CD has processed all the Applications  (and thus no need to use a sleep statement and guess how long it will take.) This leads to more reliable tests.